proof/crefine/AARCH64/Schedule_C.thy

```diff
@@ -216,13 +244,61 @@ lemma switchToThread_ccorres':
 
 lemmas word_log2_max_word_word_size = word_log2_max[where 'a=machine_word_len, simplified word_size, simplified]
 
+lemma ccorres_pre_getQueue:
+  assumes cc: "\<And>queue. ccorres r xf (P queue) (P' queue) hs (f queue) c"
+  shows   "ccorres r xf (\<lambda>s. P (ksReadyQueues s (d, p)) s \<and> d \<le> maxDomain \<and> p \<le> maxPriority)
+                        {s'. \<forall>queue. (let cqueue = index (ksReadyQueues_' (globals s'))
+                                                         (cready_queues_index_to_C d p) in
+                                      ctcb_queue_relation queue cqueue) \<longrightarrow> s' \<in> P' queue}
+           hs (getQueue d p >>= (\<lambda>queue. f queue)) c"
+  apply (rule ccorres_guard_imp2)
+  apply (rule ccorres_symb_exec_l2)
+     defer
+     defer
+     apply (rule gq_sp)
+    defer
+    apply (rule ccorres_guard_imp)
+    apply (rule cc)
+     apply clarsimp
+     apply assumption
+    apply assumption
```

Comment:
this proof looks like a struggle uphill both ways in the snow... any insight into why the rules are matching so badly? `ccorres_symb_exec_l` has 4 versions (2,3,and `'`), none of them help?

Reply:
I think this proof was copied from somewhere, and it had this rather unusual structure. Would it be better to try to rewrite this?

Reply:
I'd say it's worth a try, but as it's copied, I'd say put it on your TODO list and like the FIXME move above do it in another PR.